cpp/tests/community/louvain_test.cu

```diff
@@ -62,20 +62,32 @@ TEST(louvain, success)
 
   raft::handle_t handle;
 
-  std::tie(num_level, modularity) = cugraph::louvain(handle, G, result_v.data().get());
-
-  cudaMemcpy((void*)&(cluster_id[0]),
-             result_v.data().get(),
-             sizeof(int) * num_verts,
-             cudaMemcpyDeviceToHost);
-
-  int min = *min_element(cluster_id.begin(), cluster_id.end());
-
-  std::cout << "modularity = " << modularity << std::endl;
-
-  ASSERT_GE(min, 0);
-  ASSERT_GE(modularity, 0.402777 * 0.95);
-  ASSERT_EQ(result_v, result_h);
+  // "FIXME": remove this check once we drop support for Pascal
+  //
+  // Calling louvain on Pascal will throw an exception, we'll check that
+  // this is the behavior while we still support Pascal (device_prop.major < 7)
+  //
+  cudaDeviceProp device_prop;
+  CUDA_CHECK(cudaGetDeviceProperties(&device_prop, 0));
+
+  if (device_prop.major < 7) {
+    EXPECT_THROW(cugraph::louvain(handle, G, result_v.data().get()), cugraph::logic_error);
+  } else {
+    std::tie(num_level, modularity) = cugraph::louvain(handle, G, result_v.data().get());
+
+    cudaMemcpy((void*)&(cluster_id[0]),
+               result_v.data().get(),
+               sizeof(int) * num_verts,
+               cudaMemcpyDeviceToHost);
```

Comment:
Better use raft::update_host followed by stream sync.

Reply:
Modified all Louvain derived tests to use `raft::update_host` followed by `cudaDeviceSynchronize`